library/cipher.c

```diff
@@ -849,6 +849,9 @@ static int get_pkcs_padding(unsigned char *input, size_t input_len,
     }
 
```

Comment:
This is a robustness improvement, not a bug fix, but we might want to apply it to 3.6 LTS anyway.

Note that the `mbedtls_cipher` API may or may not still exist in the next major version of Mbed TLS, at there are no more 3.x releases planned outside of 3.6 LTS, so at this point it's unsure whether this change will ever be released.

The legacy mbedtls crypto API has traditionally not been robust in this way. In the PSA crypto API, which replaces the legacy API, we have followed a robustness guideline which does include things like setting output parameters to least-harmful values on error.

Reply:
I agree. Thanks for reviewing!